exporter/exporter.go

```diff
@@ -14,15 +15,13 @@ type Exporter interface {
 type ExporterInstance interface {
 	Name() string
 	Config() Config
-	Export(ctx context.Context, src Source, sessionID string) (map[string]string, error)
+	Export(ctx context.Context, src *Source, sessionID string) (map[string]string, error)
 }
 
 type Config struct {
 	Compression compression.Config
 }
 
-type Source struct {
-	Ref      cache.ImmutableRef
-	Refs     map[string]cache.ImmutableRef
-	Metadata map[string][]byte
-}
+type Source = result.Result[cache.ImmutableRef]
+type Attestation = result.Attestation[cache.ImmutableRef]
+type InTotoAttestation = result.InTotoAttestation[cache.ImmutableRef]
```

Comment:
this grouping might be nice:
```suggestion
type (
	Source = result.Result[cache.ImmutableRef]
	Attestation = result.Attestation[cache.ImmutableRef]
	InTotoAttestation = result.InTotoAttestation[cache.ImmutableRef]
)
```